### PR TITLE
Feature - warning when removing a linked item from a protocol

### PR DESCRIPTION
**Pull request by joefitter**
* Added helper to show confirm message when a linked objective, establishment or POLE is removed
* Replaced Objective with standard repeater item

### CODE REVIEW

File: client/helpers/index.js [hunk 1]
```diff
@@ -145,3 +145,42 @@ export const stripInvalidXmlChars = text => {
   }
   return text.replace(/([^\x09\x0A\x0D\x20-\uD7FF\uE000-\uFFFC\u{10000}-\u{10FFFF}])/ug, '')
 };
+
+// protocol 1, protocol 2 and protocol 3
```
Comment (lennym) [inline client/helpers/index.js:149]: Not sure what this comment is for, but it seems to represent neither the arguments nor return value of this function.

File: client/helpers/index.js [hunk 2]
```diff
@@ -145,3 +145,42 @@ export const stripInvalidXmlChars = text => {
...
+
+  // item doesn't appear in any protocols
+  if (!protocols.length) {
+    return true;
+  }
+
+  let message = `Protocols will be affected\n\nRemoving this ${singularName} will also remove it from`;
+
+  // item appears in all protocols
+  if (protocols.length === (project.protocols || []).length) {
```
Comment (lennym) [inline client/helpers/index.js:176]: Does this handle deleted protocols?

File: client/helpers/index.js (hunk 1 above)
Comment (joefitter) [reply client/helpers/index.js:149]: it's _one_ of the return values - I can expand it a bit

File: client/helpers/index.js (hunk 2 above)
Comment (joefitter) [reply client/helpers/index.js:176]: good point, bro###

File: client/helpers/index.js (hunk 1 above)
Comment (lennym) [reply client/helpers/index.js:150]: The whole map just results in `1, 2 and 3` though doesn't it? I don't see where "protocol" would be added.

File: client/pages/sections/repeater/index.js [hunk 3]
```diff
@@ -1,27 +1,37 @@
...
-      <Fieldset
-        fields={fields}
-        values={values}
-        prefix={prefix}
-        onFieldChange={(key, value) => updateItem({ [key]: value })}
-      />
-    </div>
-  </Fragment>
-)
+export function Item({ index, fields, values, updateItem, removeItem, length, prefix, singular, confirmRemove }) {
```
Comment (lennym) [inline client/pages/sections/repeater/index.js:10]: Can we whack some newlines in here please? It's getting pretty girthy.

File: client/pages/sections/repeater/index.js [hunk 4]
```diff
@@ -1,27 +1,37 @@
...
+  const project = useSelector(state => state.project);
+  function confirmRemoveItem(e) {
+    e.preventDefault();
+    if (confirmRemove(project, values)) {
```
Comment (lennym) [inline client/pages/sections/repeater/index.js:14]: This probably wants a check that `confirmRemove` is defined and is a function.

File: client/helpers/index.js [hunk 5]
```diff
@@ -145,3 +145,45 @@ export const stripInvalidXmlChars = text => {
...
+    return window.confirm(`${message} all protocols.`)
+  }
+
+  const protocolText = protocols.length === 1
+    // item appears in a single protocol
+    ? `protocol ${protocols[0].index + 1}`
+    // item appears in many protocols, list them
+    : `protocols ${getProtocolParts(protocols)}`;
+  return window.confirm(`${message} ${protocolText}.`);
+};
```
Comment (lennym) [inline client/helpers/index.js:189]: Could we split some of "helpers" out into individual files? Massive dumping ground index.js files are kind of a pain to work with.

Suggest creating `confirm-remove.js` and putting the green bits of this diff into it.

File: client/helpers/index.js (hunk 1 above)
Comment (joefitter) [reply client/helpers/index.js]: yeah that's what I meant, derp - I'll fix the comment

**Pull request merged**